File: android/app/src/main/kotlin/com/youxia/nova_frame/native/channel/PlatformChannel.kt
package com.youxia.nova_frame.native.channel
import android.content.Context
import android.os.Build
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class PlatformChannel(
    @Suppress("UNUSED_PARAMETER") private val context: Context,
    private val messenger: BinaryMessenger,
    private val channelName: String = CHANNEL_NAME,
) : MethodChannel.MethodCallHandler {

    private val channel: MethodChannel = MethodChannel(messenger, channelName)

    fun register() {
        channel.setMethodCallHandler(this)
    }

    fun unregister() {
        channel.setMethodCallHandler(null)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            METHOD_GET_PLATFORM_VERSION -> {
                val version = "Android ${Build.VERSION.RELEASE} (SDK ${Build.VERSION.SDK_INT})"
                result.success(version)
            }

            else -> result.notImplemented()
        }
    }

    companion object {
        const val CHANNEL_NAME = "nova/platform"
        const val METHOD_GET_PLATFORM_VERSION = "getPlatformVersion"
    }
}

File: ios/Runner/Native/Channel/PlatformChannel.swift
import Flutter
import UIKit

final class PlatformChannel {
  private static let channelName = "nova/platform"
  private static let methodGetPlatformVersion = "getPlatformVersion"

  static func register(with messenger: FlutterBinaryMessenger) {
    let channel = FlutterMethodChannel(name: channelName, binaryMessenger: messenger)
    channel.setMethodCallHandler { call, result in
      switch call.method {
      case methodGetPlatformVersion:
        let systemName = UIDevice.current.systemName
        let systemVersion = UIDevice.current.systemVersion
        result("\(systemName ) \(systemVersion)")
      default:
        result(FlutterMethodNotImplemented)
      }
    }
  }

}
